pkg/services/settings/settingsprovider/settingsprovider.go

```diff
@@ -0,0 +1,162 @@
+package settingsprovider
+
+import (
+	"reflect"
+	"sync"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/registry"
+	"github.com/grafana/grafana/pkg/services/settings"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+var (
+	logger = log.New("settingsprovider")
+)
+
+func init() {
+	registry.Register(&registry.Descriptor{
+		Name:         "SettingsProvider",
+		Instance:     &Implementation{},
+		InitPriority: registry.High - 5,
+	})
+}
+
+type settingsBag map[string]map[string]string
+
+type Implementation struct {
+	sync.RWMutex
+	FileCfg  *setting.Cfg       `inject:""`
+	SQLStore *sqlstore.SQLStore `inject:""`
+	settings settingsBag
+}
+
+func (i *Implementation) Init() (err error) {
+	i.settings, err = i.loadAndMergeSettings()
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		syncTicker := time.NewTicker(1 * time.Minute)
```

Comment:
I'm planning to introduce a new settings section for settings (😂), which will basically include options like:

- `syncTicker` duration (how often should we do checks for settings updates).
- A list of settings sections (or specific keys) allowed to be overwritten from the database.
- Any other idea?  

WDYT?? 🤔